src/tty-ask-password-agent/tty-ask-password-agent.c

```diff
@@ -313,6 +324,40 @@ static int wall_tty_block(void) {
         return fd;
 }
 
+static int parse_filter(const char *arg) {
+        const char *l;
+        bool strict = false;
+        int r;
+        pid_t p;
+
+        assert(arg || !strict);
+
+        if (!arg)
+                goto default_noarg;
+
+        if (streq(arg, "all")) {
+                arg_query_pid = 0;
+                return 1;
+        }
+
+        l = startswith(arg, "pid="); // IDEA: Support other filter values
+
+        r = parse_pid(l /* "--filter=pid=XXX" */ ?: arg /* "--filter=XXX" */, &p);
+        if (r < 0 || p <= 0) {
+                if (!strict)
+                        goto default_noarg;
+
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to parse --filter='%s'.", arg);
```

Comment:
if you have an error code (which you will if parse_pid() fails) then please pass it to the logging functions, i.e. as first arg here, instead of making up EINVAL

Reply:
should be better now :+1: